app/src/org/commcare/CommCareApplication.java

```diff
@@ -1235,102 +1224,4 @@ private void setRxJavaGlobalHandler() {
             Thread.currentThread().getUncaughtExceptionHandler().uncaughtException(Thread.currentThread(), throwable);
         });
     }
-
-    @Override
-    public void onStateChanged(@NonNull LifecycleOwner source, @NonNull Lifecycle.Event event) {
-        switch (event.name()) {
-            case "ON_RESUME":
-                Logger.log("commcare-state-changed", "User returning to CommCare. App lifecycle changed: " + event.name());
-                break;
-            case "ON_PAUSE":
-                Logger.log("commcare-state-changed", "User leaving CommCare. App lifecycle changed: " + event.name());
-                logMemoryAndBatteryInfo("commcare-state-changed");
-                break;
-        }
-    }
-
-    private void logMemoryAndBatteryInfo(String logType) {
-        try {
-            logMemoryInfo();
-            logBatteryInfo();
-        } catch (Exception e){
-            Logger.log(logType, "Exception while logging memory and battery info: "+ e.getMessage());
-        }
-    }
-
-    @Override
-    public void onTrimMemory(int level) {
-        super.onTrimMemory(level);
-        Logger.log("memory-trim-request", "Memory level: " + getMemoryLevelName(level));

```

Comment:
think there is still value in logging just this event while debugging issues so that we know that the app may have been killed in between a workflow.

Reply:
Should we put behind a custom property? this can create a lot of "noise"

Reply:
No, I don't think we should add custom properties for a small log event like this, I was mostly assuming that it shoulf not really create that much noise. Like a few events per hour per user on average seems alright to me.

Reply:
I think depending on the OS it can create a lot, here's a sample from a random day and within the same hour:
2024-08-19T06:08:26.608000Z | Memory level: TRIM_MEMORY_UI_HIDDEN
2024-08-19T06:08:40.992000Z | Memory level: TRIM_MEMORY_COMPLETE
2024-08-19T06:08:52.271000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:09:34.947000Z | Memory level: TRIM_MEMORY_BACKGROUND
2024-08-19T06:10:40.099000Z | Memory level: TRIM_MEMORY_BACKGROUND
2024-08-19T06:10:52.553000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:10:52.910000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:10:56.934000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:10:59.009000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:11:01.966000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:11:23.742000Z | Memory level: TRIM_MEMORY_BACKGROUND
2024-08-19T06:11:28.058000Z | Memory level: TRIM_MEMORY_BACKGROUND
2024-08-19T06:11:31.395000Z | Memory level: TRIM_MEMORY_BACKGROUND
2024-08-19T06:11:33.248000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:11:41.037000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:11:44.131000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:11:52.904000Z | Memory level: TRIM_MEMORY_BACKGROUND
2024-08-19T06:11:55.126000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:12:00.991000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:12:01.944000Z | Memory level: TRIM_MEMORY_MODERATE
2024-08-19T06:25:01.135000Z | Memory level: TRIM_MEMORY_UI_HIDDEN
2024-08-19T06:26:49.159000Z | Memory level: TRIM_MEMORY_COMPLETE
2024-08-19T06:28:55.817000Z | Memory level: TRIM_MEMORY_COMPLETE
2024-08-19T06:29:56.910000Z | Memory level: TRIM_MEMORY_UI_HIDDEN
2024-08-19T06:30:21.598000Z | Memory level: TRIM_MEMORY_COMPLETE
2024-08-19T06:33:10.229000Z | Memory level: TRIM_MEMORY_UI_HIDDEN
2024-08-19T06:33:13.086000Z | Memory level: TRIM_MEMORY_COMPLETE
2024-08-20T06:34:24.137000Z | Memory level: TRIM_MEMORY_UI_HIDDEN
2024-08-20T06:49:15.735000Z | Memory level: TRIM_MEMORY_COMPLETE
2024-08-20T06:49:38.745000Z | Memory level: TRIM_MEMORY_COMPLETE
2024-08-20T06:49:39.845000Z | Memory level: TRIM_MEMORY_COMPLETE

Reply:
got it, is it for the same user ? If so, agree it may be a lot. 

How about we oveeride `onDestroy` in the application class and log application destory. That should be useful and conscise and will probably give us the same info that app was killed (can be manually or by OS)

Reply:
> got it, is it for the same user ? If so, agree it may be a lot.

Yes, same user. It's not consistent, so you get periods of relative silence, with just a handful of request in an hour, but it's hard to predict the volume of logs this will generate.

> How about we oveeride onDestroy in the application class and log application destory. 
We can do that, but I doubt `onDestroy` is called when the app is forcefully killed.